test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -9481,7 +9483,42 @@ def create_import_hook(self, name, attrs=None, body=None, overwrite=True):
 
         # In 12.0 A MoM hook must be enabled and the event set prior to
         # importing, otherwise the MoM does not get the hook content
-        return self.import_hook(name, body)
+        ret = self.import_hook(name, body)
+
+        # In case of mom hooks, make sure that the hook related files
+        # are successfully copied to the MoM
+        try:
+            if attrs['event'].startswith('exec'):
+                hook_py = name + '.PY'
+                hook_hk = name + '.HK'
+                pyfile = os.path.join(self.pbs_conf['PBS_HOME'],
+                                      "server_priv", "hooks", hook_py)
+                hfile = os.path.join(self.pbs_conf['PBS_HOME'],
+                                     "server_priv", "hooks", hook_hk)
+                logmsg = hook_py + ";copy hook-related file request received"
+
+                cmd = os.path.join(self.client_conf['PBS_EXEC'], 'bin',
+                                   'pbsnodes') + ' -a'
+                cmd_out = self.du.run_cmd(self.hostname, cmd, sudo=True)
+                if cmd_out['rc'] == 0:
+                    for i in cmd_out['out']:
+                        if re.search('Mom =', i):
```

Comment:
Since search() will match anywhere on the line, this might match a string resource with 'Mom =' in it (it's unlikely, but possible).  I'd use re.match(r'\s+Mom =').  re.match() searches from the start of the line.